cmd/kubectl-k8ssandra/register/command.go

```diff
@@ -0,0 +1,73 @@
+package register
+
+import (
+	"fmt"
+
+	"github.com/charmbracelet/log"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+var RegisterClusterCmd = &cobra.Command{
+	Use:   "register [flags]",
+	Short: "register a data plane into the control plane.",
+	Long:  `register creates a ServiceAccount on a source cluster, copies its credentials and then creates a secret containing them on the destination cluster. It then also creates a ClientConfig on the destination cluster to reference the secret.`,
+	Run:   entrypoint,
+}
+
+func SetupRegisterClusterCmd(cmd *cobra.Command, streams genericclioptions.IOStreams) {
+	RegisterClusterCmd.Flags().String("source-kubeconfig",
+		"",
+		"path to source cluster's kubeconfig file - defaults to KUBECONFIG then ~/.kube/config")
+	RegisterClusterCmd.Flags().String("dest-kubeconfig",
+		"",
+		"path to destination cluster's kubeconfig file - defaults to KUBECONFIG then ~/.kube/config")
+	RegisterClusterCmd.Flags().String("source-context", "", "context name for source cluster")
+	RegisterClusterCmd.Flags().String("dest-context", "", "context name for destination cluster")
```

Comment:
**Suggestion:** could we make these arguments instead of flags?

I feel like something along the lines of `kubectl k8ssandra register dest-context src-context` would be easier to work with, given these flags are mandatory anyway. It'll shorten the commands.

Reply:
I'd rather not do this actually. I think there are quite a complicated set of arguments to this command between contexts (which is language we have made up that doesn't quite match the k8s usage...), namespaces, destination resource names, etc. 

I'd rather have each of these very clearly spelt out via flags, as I think this will make things less confusing (if more verbose) over the longer term. We should provide tab completion if we are worried about usability.